Feat. #12 메인페이지 기능 구현

Mark3891 opened the pull request:
<!--
🙏 PR 제목 컨벤션 (Gitmoji + 타입 + 이슈 번호 + 작업 요약)
예시: ✨ Feat: #167 예약 취소 구현
※ PR 생성 시 Assignees 및 Labels 설정도 잊지 마세요!
-->

## ✨ What’s this PR?
### 📌 관련 이슈 (Related Issue)
<!-- 해당 PR이 어떤 이슈를 해결하는지 연결해주세요 -->
- Closes #12 
- Closes #33 

---

### 🧶 주요 변경 내용 (Summary)
<!-- 이번 PR에서 작업한 핵심 변경 사항을 작성해주세요 -->

✨ Feat. 메인 페이지 기능적으로 구현

- LocationManager를 활용하여 유저의 현재 위치를 받아오고, reverseGeocodeLocation으로 주소를 추출했습니다.

- Haversine 공식을 사용해 반경 내 Mote 데이터를 필터링합니다.

- AR 뷰 전환 버튼을 추가했습니다.

- 임시 MockData 및 전체 데이터 모델을 구성했습니다.

✨ Feat. 메인 화면에 현재 위치 지도(Map) 추가

- SwiftUI의 Map 컴포넌트를 사용해 현재 위치를 시각적으로 확인할 수 있도록 지도 뷰를 추가했습니다.

- 현재 위치 마커를 표시하기 위한 IdentifiableLocation 구조체를 정의했습니다.

✨ Feat. Half 모달 시트 드래그 시 Full 모달 전환 구현

- DragGesture를 활용해 시트 드래그 시 위치에 따라 .half ↔ .full 상태 전환이 가능하도록 UX 개선을 진행했습니다.

♻️ Refactor. BottomSheet에 X 버튼으로 Half 전환 로직 분리

- MyRecordBottomSheet가 @Binding을 통해 외부 상태를 제어할 수 있도록 구조를 개선했습니다.

- .full 상태에서만 나타나는 X 버튼을 통해 .half 상태로 전환할 수 있도록 구현했습니다.

---

### 📸 스크린샷 (Optional)
<!-- UI 작업의 경우, 구현한 화면을 첨부해주세요 -->
<!-- 이미지 크기 조절 예시: <img width="300" alt="설명" src="링크"> -->
<img width="300" alt="image" src="https://github.com/user-attachments/assets/0441da8e-3476-4e0e-a1e0-abfed6ebbecd" />
<img width="300" alt="image" src="https://github.com/user-attachments/assets/898ade14-fbbb-4a4a-8954-c59a4c5e1844" />



---

### 🧪 테스트 / 검증 내역
<!-- 동작 확인 여부나 시나리오 테스트 내용을 간단히 써주세요 -->

- [x] UI 정상 동작 확인
- [x] iPhone 15, iOS 17.4 환경에서 정상 동작
- [ ] 다크모드 테스트는 이후 이슈로 분리 예정 (#000)

---

### 💬 기타 공유 사항
<!-- 리뷰어가 참고하면 좋을 정보, 고민했던 지점 등을 적어주세요 -->

- mockData와 임시 데이터 모델을 만들어놔서 헷갈릴 수 있어요 추후에 정리 할게요
- UI 작업할 때 Component화 하겠습니다.

---

### 🙇🏻‍♀️ 리뷰 가이드 (선택)
<!-- 리뷰어가 중점적으로 보면 좋을 포인트가 있다면 알려주세요 -->

- mainModel은 MainView에서 쓰기 위한 데이터이고, MockData는 전체 데이터 type을 다 갖춘 상태여서 따로 다 갖춘 데이터 모델을 만들어 두었습니다.

Mark3891 committed:
✨ Feat. 메인 페이지 기능적으로 구현

✨ Feat. 메인 페이지 기능적으로 구현

Why: 유저가 앱을 실행했을 때 현재 위치를 확인하고,
  그 주변에 과거에 기록한 Mote가 있는지 확인할 수 있어야 함

How:
- LocationManager를 통해 현재 위치 가져오기
- reverseGeocodeLocation으로 주소 문자열 추출
- Haversine 거리 계산으로 반경 내 모트 필터링
- ARCameraView로 이동하는 버튼 추가
- 임시 전체 DataModel 구현 및 MockData 구현
Tag: #swift #location #corelocation #ui #feature

Mark3891 committed:
UI. Half 모달 시트 드래그 시 Full 모달 전환 구현

✨ . Half 모달 시트 드래그 시 Full 모달 전환 구현

Why: 사용자 경험을 개선하기 위해 Half 모달을 위로 드래그하면
     Full 모달로 자연스럽게 확장되도록 처리 필요

How:
  - DragGesture의 수직 이동값을 기준으로 상태 전환 로직 구현
  - threshold 기준 100pt 이상 드래그 시 .full 또는 .half 상태로 전환
  - sheet offset 계산 시 y 값이 음수로 가지 않도록 max 적용
  - 기존 커스텀 BottomSheet 구조 유지하며 제스처 기반 UX 개선

Tag: #ui #swiftui #sheet #drag #modal #interaction

Mark3891 committed:
♻️ Refactor. BottomSheet에 X 버튼으로 Half 전환 로직 분리

♻️ Refactor. BottomSheet에 X 버튼으로 Half 전환 로직 분리

Why: 시트 전환 로직을 명확히 분리하고, 모듈 간 상태 전달 구조 개선 필요

How:
  - `MyRecordBottomSheet`에서 `@Binding`을 사용해 외부 상태 제어 가능하게 수정
  - `.full` 상태에서만 표시되는 `X` 버튼 추가
  - 버튼 클릭 시 `.half` 상태로 전환되도록 처리
  - MainView에서 `sheetPosition`을 바인딩으로 전달

Tag: #ui #swiftui #refactor #bottomsheet #binding

Mark3891 committed:
🎨 Style. 코드 정리 및 불필요한 공백/줄바꿈 제거

🎨 Style. 코드 정리 및 불필요한 공백/줄바꿈 제거

Why: 가독성을 높이고 코드 스타일 일관성을 유지하기 위함

How:
  - 파일 전반에 걸쳐 불필요한 공백, 빈 줄, 들여쓰기 수정
  - 의미 없는 줄 간 간격 제거 및 들여쓰기 재정렬
  - 로직 변경 없이 스타일 정리만 수행

Tag: #style #cleanup #swift #refactor

Mark3891 committed:
✨ Feat. 메인 화면에 현재 위치 지도(Map) 추가

✨ Feat. 메인 화면에 현재 위치 지도(Map) 추가

Why: 사용자가 자신의 현재 위치를 시각적으로 확인할 수 있도록 하기 위함

How:
  - `MapKit`과 SwiftUI의 `Map` 컴포넌트를 활용해 지도 뷰 추가
  - `LocationManager`에서 가져온 현재 위치를 기반으로 마커 표시
  - `IdentifiableLocation` 구조체 생성하여 Marker에 바인딩 가능하도록 처리
  - `.cornerRadius` 및 `.frame` 지정으로 UI 정리

Tag: #swiftui #mapkit #location #feature #ui #map

Mark3891 committed:
🔥 Remove. MainView에서 Map 관련 코드 제거

Why:
이번 구현에서는 지도를 화면에 표시하지 않기로 결정되어, MapKit 및 관련 코드가 불필요해졌기 때문입니다.

How:

MapKit import 제거

사용하지 않는 Map 컴포넌트 관련 코드 제거

지도 뷰 없이도 위치 기반 기능이 동작하도록 구조 유지

Tag:
#remove #swiftui #mapkit #cleanup

freeskyES commented on Packages/Features/Sources/Features/Main/MainView.swift:
```diff
@@ -5,13 +5,86 @@
+                "현재 위치:",
+                location.coordinate.latitude,
+                location.coordinate.longitude
+            )
```
소스가 길어지는 이부분 별도 함수로 분리해주시면, 가독성이 더 좋아질 것 같아요!
```
.onReceive(locationManager.$currentLocation.compactMap { $0 }) { location in
    handleLocationUpdate(location)
}
```

NOP-YA commented:
PR 양식을 준수해주세요 ㅜㅜㅠ

freeskyES reviewed (approved):
고생하셨어요!! 잘 나눠놓으셔서 금방 읽혔어요 👍

freeskyES committed:
Fix. Resolve conflict in MainView.swift

Why:
- Integrate recent changes from develop branch.

How:
- Manually resolved conflicts in MainView.swift.
